packages/node/src/starknet/block.starknet.ts

```diff
@@ -86,8 +86,9 @@ export function filterTransactionsProcessor(
     if (filter.function) {
       const index = decodedCalls?.findIndex(
         (call) =>
-          hexEq(call.selector, filter.function!) ||
-          hexEq(call.selector, encodeSelectorToHex(filter.function!)),
+          typeof filter.function === 'string' &&
```

Comment:
Can `filter.function` be another type? Would a `!!filter.function &&` be more suitable?